### PR TITLE
Create sad path error messages for users

### PR DESCRIPTION
**Pull request by taranveersingh93**
Description
- Created relevant error displays for if our API's fail. 
- The error for PATCH is only in the console because that does not affect user's experience but will be of interest for the developer.
- The error for POST and DELETE are being displayed both on the grid and the modal. Earlier, user got a feedback confirming happy path. Now user gets sad path feedback too. 
- Our functions are stringent with conditionals so we've already prevented all imaginable sad paths. So for this PR's purpose: consider any path that does not get a success status message from our server. 
- Adding to the stringent conditions, our happy path functions are ONLY going to be invoked if the status message in the response confirms success (I referred to the response messages and checked for expected substrings in success). 
- All tests still passing

Testing Notes
- Load the web app and once loaded, close the server from your terminal. Then try adding/deleting a recipe. The add/minus should not change and relevant feedback should appear. The recipe should not appear in saved recipes either.
- Try the same again from within the modal (close the server only once you're inside the modal view. After closing the server, try bookmarking the recipe).
- Try both adding/deleting
- Try from grid and modal
- Try from different screen sizes.

### CODE REVIEW

**Comment by taranveersingh93** [comment]
[[extension 5] Error handling audit](https://trello.com/c/GKVP9hBm/57-extension-5-error-handling-audit)

File: src/apiCalls.js [hunk 1]
```diff
@@ -108,20 +117,35 @@ const updateCurrentUser = (user) => {
...
+      }
     })
-    .catch(err => console.error(err));
+    .catch(err => {
+      showError(recipeID);
+      console.error(err);
+    });
 }
 
 const deleteRecipeToCook = (userID, recipeID, e) => {
```
Comment (lauraguerra1) [inline src/apiCalls.js]: I just realized these to functions are almost identical. Can we dry this up with the following dynamic function replacing them? 

```
const updateServerRecipe = (userID, recipeID, e, requestType) => {
  const conditions = {
    "POST": 'added',
    "DELETE": 'removed'
  }
  updateRecipe(userID, recipeID, requestType)
  .then((res) => res.json())
  .then(status => {
    if (status.message.includes(`was ${conditions[requestType]}`)) {
      getUsersAfterUpdate(userID, recipeID, e);
    } else {
      showError(recipeID)
    }
  })
  .catch(err => {
    showError(recipeID);
    console.error(err);
  });
}
```

"POST" or "DELETE" would be passed in as the request type upon invokation

**Review by lauraguerra1 — CHANGES_REQUESTED** [review]

File: src/apiCalls.js (hunk 1 above)
Comment (taranveersingh93) [reply src/apiCalls.js]: Done

**Review by lauraguerra1 — APPROVED** [review]

**Review by mbenfowler — CHANGES_REQUESTED** [review]
I can't get a screen because it happens so fast, but on the recipe view the error message is word-wrapped - was this intentional?  I think it may look better all on one line.

**Review by mbenfowler — APPROVED** [review]

**Pull request merged**